Python/crossinterp.c

```diff
@@ -439,6 +442,51 @@ _PyObject_GetXIData(PyThreadState *tstate,
     return 0;
 }
 
+int
+_PyObject_GetXIData(PyThreadState *tstate,
+                    PyObject *obj, _PyXIData_t *xidata)
+{
+    return _get_xidata(tstate, obj, _PyXIDATA_XIDATA_ONLY, xidata);
+}
+
+int
+_PyObject_GetXIDataWithFallback(PyThreadState *tstate,
```

Comment:
Could we rename the long name to `_PyObject_GetXIData`?  I feel the existing `_PyObject_GetXIData` could be longer (e.g. `_PyObject_GetXIDataNoFallback`), as it is unpopular at all in newer PRs.

Reply:
I'm not sure whether the `fallback` parameter is needed.  Does this function support the following?
  * call `getdata.basic()` under the `_PyXIDATA_FULL_FALLBACK` case
  * call `getdata.fallback()` under the `_PyXIDATA_XIDATA_ONLY` case

Reply:
Good point about the function name.  I'll take a look.

As to the fallback parameter, it does not determine which function is called.  The `_PyXIData_getdata_t` functions will be called for any of the fallback values.  It's just a matter of if the fallback parameter is propagated to the getdata func.